src/transaction_builder.h

```diff
@@ -29,10 +30,12 @@
 class OrchardWallet;
 namespace orchard { class UnauthorizedBundle; }
 
-uint256 ProduceZip244SignatureHash(
+uint256 ProduceShieldedSignatureHash(
+    uint32_t consensusBranchId,
     const CTransaction& tx,
     const std::vector<CTxOut>& allPrevOutputs,
-    const orchard::UnauthorizedBundle& orchardBundle);
+    const sapling::UnauthorizedBundle& saplingBundle,
+    const std::optional<orchard::UnauthorizedBundle>& orchardBundle);
```

Comment:
Ok, I think I do now – the sapling one _is_ optional, it’s just that the actual bundle is inside this struct, and it always carries a proving context even if there’s no actual bundle.

The orchard one was always optional, we just didn’t call this if we didn’t have an Orchard bundle (but that’s not possible now, since we may have either or both Sapling & Orchard (previously the Sapling version was handled elsewhere).

So, why doesn’t the orchard bundle have a similar structure, with a proving context being carried regardless of the internal bundle?